zpay32/invoice.go

```diff
@@ -288,24 +288,17 @@ func Decode(invoice string) (*Invoice, error) {
 	}
 
 	// The next characters should be a valid prefix for a segwit BIP173
-	// address. This will also determine which network this invoice is
-	// meant for.
-	var net *chaincfg.Params
-	if strings.HasPrefix(hrp[2:], chaincfg.MainNetParams.Bech32HRPSegwit) {
-		net = &chaincfg.MainNetParams
-	} else if strings.HasPrefix(hrp[2:], chaincfg.TestNet3Params.Bech32HRPSegwit) {
-		net = &chaincfg.TestNet3Params
-	} else if strings.HasPrefix(hrp[2:], chaincfg.SimNetParams.Bech32HRPSegwit) {
-		net = &chaincfg.SimNetParams
-	} else {
+	// address that match the active network.
+	if !strings.HasPrefix(hrp[2:], net.Bech32HRPSegwit) {
 		return nil, fmt.Errorf("unknown network")
 	}
 	decodedInvoice.Net = net
 
-	// Optionally, if there's anything left of the HRP, it encodes the
+	// If the HRP is longer than "ln" + the segwit prefix, parse the
```

Comment:
"Optionally, if there's anything left of the HRP after "ln" + the segwit prefix, it encodes the payment amount."